File: WinForms/CS/TreeMapMainDemo/DemosRegistration.cs
using System.Windows.Forms;
using DevExpress.DXperience.Demos;
using DevExpress.XtraEditors;

namespace DevExpress.XtraTreeMap.Demos {
    public class DemosInfo : ModulesInfo {
        public static void DoShowModule(string name, GroupControl group, RibbonMenuManager manager) {
            ModuleInfo item = DemosInfo.GetItem(name);
            Cursor currentCursor = Cursor.Current;
            Cursor.Current = Cursors.WaitCursor;
            try {
                Control oldTutorial = null;
                if(Instance.CurrentModuleBase != null) {
                    if(Instance.CurrentModuleBase.Name == name) return;
                    oldTutorial = Instance.CurrentModuleBase.TModule;
                }
                Instance.CurrentModuleBase = item;
                TutorialControlBase tutorialBase = item.TModule as TutorialControlBase;
                tutorialBase.Bounds = group.DisplayRectangle;
                tutorialBase.Visible = false;
                group.Controls.Add(tutorialBase);
                tutorialBase.Dock = DockStyle.Fill;
                tutorialBase.RibbonMenuManager = manager;
                tutorialBase.TutorialName = name;
                tutorialBase.Visible = true;
                item.WasShown = true;
                if(oldTutorial != null)
                    oldTutorial.Visible = false;
                TreeMapTutorialControl treeMapTutorial = tutorialBase as TreeMapTutorialControl;
                if(treeMapTutorial != null) {
                    manager.AllowExport(treeMapTutorial.ActiveDemoModule);
                }
            }
            catch { }
            finally { Cursor.Current = currentCursor; }
            RaiseModuleChanged();
        }
    }
}

File: WinForms/VB/SankeyMainDemo/DemosRegistration.vb
Imports System.Windows.Forms
Imports DevExpress.DXperience.Demos
Imports DevExpress.XtraEditors

Namespace DevExpress.XtraSankey.Demos

    Public Class DemosInfo
        Inherits ModulesInfo

        Public Shared Sub DoShowModule(ByVal name As String, ByVal group As GroupControl, ByVal manager As RibbonMenuManager)
            Dim item As ModuleInfo = GetItem(name)
            Dim currentCursor As Cursor = Cursor.Current
            Cursor.Current = Cursors.WaitCursor
            Try
                Dim oldTutorial As Control = Nothing
                If Instance.CurrentModuleBase IsNot Nothing Then
                    If Equals(Instance.CurrentModuleBase.Name, name) Then Return
                    oldTutorial = Instance.CurrentModuleBase.TModule
                End If

                Instance.CurrentModuleBase = item
                Dim tutorialBase As TutorialControlBase = TryCast(item.TModule, TutorialControlBase)
                tutorialBase.Bounds = group.DisplayRectangle
                tutorialBase.Visible = False
                group.Controls.Add(tutorialBase)
                tutorialBase.Dock = DockStyle.Fill
                tutorialBase.RibbonMenuManager = manager
                tutorialBase.TutorialName = name
                tutorialBase.Visible = True
                item.WasShown = True
                If oldTutorial IsNot Nothing Then oldTutorial.Visible = False
                Dim sankeyTutorial As SankeyTutorialControl = TryCast(tutorialBase, SankeyTutorialControl)
                If sankeyTutorial IsNot Nothing Then manager.AllowExport(sankeyTutorial.SankeyDiagramControl)
            Catch
            Finally
                Cursor.Current = currentCursor
            End Try

            Call RaiseModuleChanged()
        End Sub
    End Class
End Namespace
